apps/meteor/app/api/server/v1/oauthapps.ts

```diff
@@ -48,6 +54,24 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'oauth-apps.info',
+	{ authRequired: true, validateParams: isOauthAppsInfoParams },
+	{
+		async get() {
+			const oauthApp = await OAuthApps.findOneAuthAppByIdOrClientId(this.queryParams);
```

Comment:
Add a projection since you're returning just clientId/name

Reply:
(this will imply updating the model typings for allowing the projection to go as the current method doesn't allow it)

Reply:
Here, unfortunately, I encounter a problem and need help or a hint to solve this comment.
As I see, the method `findOneAuthAppByIdOrClientId` is [declared](https://github.com/RocketChat/Rocket.Chat/blob/16a9c862d20d93c614ce76801f29d03f41b98abe/packages/model-typings/src/models/IOAuthAppsModel.ts#L7) in the `IOAuthAppsModel` interface and [returns](https://github.com/RocketChat/Rocket.Chat/blob/16a9c862d20d93c614ce76801f29d03f41b98abe/packages/model-typings/src/models/IOAuthAppsModel.ts#L14) a `Promise<IOAuthApps>`.

The `IOAuthAppsModel` interface, in turn, [extends](https://github.com/RocketChat/Rocket.Chat/blob/16a9c862d20d93c614ce76801f29d03f41b98abe/packages/model-typings/src/models/IOAuthAppsModel.ts#L6) the `IBaseModel` interface, which takes a type as an argument that, in turn, [must extend](https://github.com/RocketChat/Rocket.Chat/blob/16a9c862d20d93c614ce76801f29d03f41b98abe/packages/model-typings/src/models/IBaseModel.ts#L42) the type `{ _id: string }`. This definition fits `IOAuthApps`, but does not fit `OAuthAppsInfo`.

How can I correctly modify the call `const oauthApp = await OAuthApps.findOneAuthAppByIdOrClientId(this.queryParams);` to return `OAuthAppsInfo` instead of `IOAuthApps`?

Reply:
I added projection to the `findOneAuthAppByIdOrClientId` method of the `IOAuthAppsModel` and `OAuthAppsRaw` models. I hope I changed the code everywhere it was needed and didn't forget anything.